torch/_dynamo/decorators.py

```diff
@@ -183,6 +197,11 @@ def mark_dynamic(t, index):
     before torch.compile.
 
     """
+    if is_traceable_wrapper_subclass(t):
+        # default behavior: mirror mark_dynamic() on all inner tensors with same dim as t
+        # TODO: Make this configurable via a supported public API
+        _apply_func_to_inner_tensors_of_same_dim(mark_dynamic, t, index)
```

Comment:
Just a nit / eventual TODO: once we have `__tensor_mark_dynamic__`, we'll need to explicitly make sure to NOT do this if a subclass has implemented it.